core-processor/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -2087,6 +2099,32 @@ private int pushGetValueForPathCall(GeneratorAdapter injectMethodVisitor, ClassE
         return optionalInstanceIndex;
     }
 
+    private boolean pushValueBypassingBeanContext(GeneratorAdapter writer, ClassElement type) {
```

Comment:
Is the purpose of this shortcut just optimization, or does it have a behavioral difference vs `context.getBean(ConversionService.class)`? If the purpose of this is behavioral, it seems clearer to just do `beanContext.getConversionService()` where it's necessary, instead of changing the codegen

Reply:
Well, we have only one `ConversionService` (resolved exactly by this type) so if you want to inject it, there is no point in going thru the bean context resolution.

Reply:
so it's just an optimization? in that case i think it's fine.

my concern was that if this is a behavior difference (e.g. getBean doesnt work early in initialization or something), then this code would have to be expanded for every type where a similar case comes up.

Reply:
Just optimization